src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs

```diff
@@ -39,96 +39,159 @@ public CommandLineConfigurationProvider(IEnumerable<string> args, IDictionary<st
-                        if (_switchMappings != null && _switchMappings.TryGetValue(keySegment, out string mappedKeySegment))
-                        {
-                            key = mappedKeySegment;
-                        }
-                        // If the switch starts with a single "-" and it isn't in given mappings , it is an invalid usage
-                        else if (keyStartIndex == 1)
-                        {
-                            throw new FormatException(SR.Format(SR.Error_ShortSwitchNotDefined, currentArg));
-                        }
-                        // Otherwise, use the switch name directly as a key
-                        else
-                        {
-                            key = currentArg.Substring(keyStartIndex, separator - keyStartIndex);
-                        }
-
-                        value = currentArg.Substring(separator + 1);
-                    }
+            Data = data;
+        }
+
+        /// <summary>
+        /// Reconcile two consecutive arguments into a key-value pair for the first.
+        /// </summary>
+        /// <remarks>
+        /// Helper function to reduce repeated code in <see cref="Load"/> method.
+        /// </remarks>
+        /// <param name="previousArg">The first string argument</param>
+        /// <param name="currentArg">The second string argument</param>
+        /// <param name="pair">
+        /// A properly resolved configuration key-value pair, or null if previous argument is invalid.
+        /// </param>
+        /// <returns>
+        /// True if the args can be resolved to a proper configuration key-value pair.
+        /// </returns>
+        private bool TryProcessArgs(string previousArg, string currentArg, out (string key, string value) pair)
+        {
+            string key, value;
+            int keyStartIndex = 0;
 
-                    // Override value when key is duplicated. So we always have the last argument win.
-                    data[key] = value;
+            if (previousArg.StartsWith("--"))
+            {
+                keyStartIndex = 2;
+            }
+            else if (previousArg.StartsWith("-"))
+            {
+                keyStartIndex = 1;
+            }
+            else if (previousArg.StartsWith("/"))
+            {
+                // "/SomeSwitch" is equivalent to "--SomeSwitch" when interpreting switch mappings
+                // So we do a conversion to simplify later processing
+                previousArg = $"--{previousArg.Substring(1)}";
+                keyStartIndex = 2;
+            }
+
+            int separator = previousArg.IndexOf('=');
+
+            if (separator < 0)
+            {
+                // If there is neither equal sign nor prefix in current argument, it is an invalid format
+                if (keyStartIndex == 0)
+                {
+                    // Ignore invalid formats
+                    pair = (null, null);
```

Comment:
micro nit: we can simplify it as `pair = default;` (as a bonus, it avoids [two `ldnull`](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhKgbyrs7oAcpsA3AIYYYdYBAi46AFSgBPAApQIYGAGc1AQSgBzNQHVsGABYARGADNBAV1wYAFIVIAGHrH7YI1rbrQMXdGDWULAAdhjaOn5eGHSOAQDWMHJ+Tq5CuNYwAJQ8gthQ2Rxc7JRc5XkFdAC8dAAmljZ2ANzFFQwA7HRWuGowrWVcAL5tdKO8AsKi4pIy8koq6j56hiYAcra4G7i48a68MB5ey6kBQSEw4ZHR1rF7dEkp/umCmTmVhaOl7Zzc+VA1OKhTZ+YE7bIDH6dbqvPqQ4ajcZ8IQiMQSKSyRTKVQaSIGIzGADSyQAaq8svcDkdvNdnoFgmEIr46DE6Y8brE0nQMlkioNON8fo9AQ0rLYMPD2jzRLVRU0JaNyoQuj04aMRpQkZNUfd2XTpbkFjjlviTNJrNxcDAAEowDDBUKU9yeGnMrnnRmRPnlQXtZX1RriyV0DValGiQgoOjW6yhUiaezekqKrjYCxxTFGpZ41ZmQN2ewAIgsEkLfkLwEEUDLLNucS5eq5Bo+2VyrBD6sR/Lc2ojUZjcYAQomvinOGmM/NsdndKbjNstpsiyWIDWK1Wa6zdclTi83rk/gVW3R2xrymeuGGpgx+7HiCPu76KhP7Jnp7jZ7niXIyW9l6Xy0rasOTZHda05AIDTbDtuwvTgr1RSNo1jGgHx9Md60SMCm3Jd5DygehaizD8VgJc1LRtO0HX/VdAI3CEMJffD6AAQllfMMGguCYKGIA) IL instructions and [codegen](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYRoG8aHuGAHKAJYA3bBhgNgECABsGAFSgBPAApQIYGLlwBBKAHNcAdQEYAFgBEYAM2wBXaRgAUxcgAY+sIQIi2d+tExuDGC2ULAAdhi6egE+GAzOQQDWMIoBLu4i0rYwAJR82AJQuVw8nNQ8lQVFDAC8DAAm1nYOANylVUwA7Aw20rgw7RU8AL4dDOP8wqLikjLySqrqmn4GxmYAcvbSW9LSie78MF4+q+lBIWEwkdGxtvEHDClpgZnY2XnVxePlndy8hSgdQS4W2AVBe1yQz+3V67wG0NG40mghEYgkUlkChUag0WmiRhMpgA0qkAGrvHKPI4nXy3V7BUIRKL+BhxBnPO7xDIMLI5ErDbi/P7PYFNGz2DCIzp88T1cUtKXjSrEHp9BHjMbUFHTdGPTkM2X5JZ41aEsxyWy8aQwABKMAwoXC1M83jprJ5l2Z0QFlWFnVVjWakulDC1OrR4mIKAYtts4XI2kcvrKyp4AisCWxJpWBPWFmDDkcACIrFJiwFi8BsFAK2z7gkeQaeUavrl8uww5rkYKPLqozG4wmAELJn5p7gZrOLXG5/Tm0y7HbbEtliB1qs1uvs/Wpc5vD75AFFdsMTtayoXngRmZMQfx0hj3v+qpTxzZ2f4+f50mKCkfVdy0ratay5Dk93rbkgiNDsu17K9uBvdFo1jeM6CfP0J0bZIIJbSlPmPKBGHqHMvzWIlLWtO0HSdQD12ArcoSwt9CMYABCeVCwwWCELgkYgA) is vectorized; `vmovdqu` instead of multiple `mov`s)

Reply:
Pushed! :)